backend/sirius-web-spring-collaborative/src/main/java/org/eclipse/sirius/web/spring/collaborative/api/ChangeDescription.java

```diff
@@ -33,12 +35,18 @@
 
     private final IInput input;
 
+    private final Map<String, Object> parameters = new HashMap<>();
+
     public ChangeDescription(String kind, UUID sourceId, IInput input) {
         this.kind = Objects.requireNonNull(kind);
         this.sourceId = Objects.requireNonNull(sourceId);
         this.input = Objects.requireNonNull(input);
     }
 
+    public Object addParameter(String key, Object value) {
+        return this.parameters.put(key, value);
+    }
```

Comment:
Do not create such stateful method, we need to keep immutable classes immutable. instead ask for a Map<> in the constructor. You can create an additional constructor which require this map and make the existing one fallback to the new constructor with an empty map.